### PR TITLE
fix: Call UIDevice methods on the main thread

### LINKED ISSUE #2282: EXC_BREAKPOINT [SentryNetworkTrackingIntegration swizzleURLSessionTask] threading violation: expected the main thread
### Platform

iOS

### Installed

Swift Package Manager

### Version

7.27.1

### Steps to Reproduce

No exact pattern to reproduce but it happens while using background thread for networking

### Expected Result

No crash

### Actual Result

screenshot

### PR DESCRIPTION
**Pull request by kevinrenskers**
## :scroll: Description

Call UIDevice methods on the main thread. This prevents an edge case crash, when the `SentryUIDeviceWrapper` gets dealloc'ed on a background thread.

## :bulb: Motivation and Context

Fixes #2282

## :green_heart: How did you test it?

Unit test.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Comment by kevinrenskers** [comment]
> I think tests for this edge case would be nice though. We could allocate and use the SentryUIDeviceWrapper on a background thread to reproduce the issue.

How would you reproduce it though? It crashes, can't catch that. We could introduce a wrapper for a wrapper but I really don't think it's worth it to be honest.

I don't even get how this person got SentryUIDeviceWrapper to be used on a background thread.

**Comment by kevinrenskers** [comment]
Now that we're using SentryDispatchQueueWrapper anyway, I added a test to make sure we call these methods using it. I also took the time to make the `TestSentryDispatchQueueWrapper` methods more consistent.

File: Sources/Sentry/include/SentryDispatchQueueWrapper.h [hunk 1]
```diff
@@ -11,7 +11,9 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (void)dispatchAsyncWithBlock:(void (^)(void))block;
 
-- (void)dispatchOnMainQueue:(void (^)(void))block;
+- (void)dispatchAsyncOnMainQueue:(void (^)(void))block;
+
+- (void)dispatchSyncOnMainQueue:(void (^)(void))block;
```
Comment (armcknight) [inline Sources/Sentry/include/SentryDispatchQueueWrapper.h:16]: Much better, thanks 👍🏻

**Review by armcknight — APPROVED** [review]

**Comment by philipphofmann** [comment]
Thank you @kevinrenskers.

**Pull request merged**